### PR TITLE
Copy, Rename and delete FileAction

### PR DESCRIPTION
**Pull request by qiyuangong**
* CopyFileAction
* RenameFileAction
* DeleteFileAction

### CODE REVIEW

File: smart-actions/src/main/java/org/smartdata/actions/hdfs/DeleteFileAction.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  private boolean deleteFile(String filePath) throws IOException, ActionException {
+    if (filePath.startsWith("hdfs")) {
+      //delete in remote cluster
+      // TODO read conf from file
+      Configuration conf = new Configuration();
+      //get FileSystem object
+      FileSystem fs = FileSystem.get(URI.create(filePath), conf);
+      if (!fs.exists(new Path(filePath))){
```
Comment (littlezhou) [inline smart-actions/src/main/java/org/smartdata/actions/hdfs/DeleteFileAction.java]: add a blank before  {

File: smart-actions/src/main/java/org/smartdata/actions/hdfs/RenameFileAction.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      // TODO read conf from files
+      //check the file name
+      if (!URI.create(dest).getHost().equals(URI.create(src).getHost())) {
+        throw new ActionException("the file names are not in the same cluster");
+      }
+      Configuration conf = new Configuration();
+      //get FileSystem object
+      FileSystem fs = FileSystem.get(URI.create(dest), conf);
+      return fs.rename(new Path(src), new Path(dest));
+    } else if (!dest.startsWith("hdfs") && !src.startsWith("hdfs")) {
```
Comment (littlezhou) [inline smart-actions/src/main/java/org/smartdata/actions/hdfs/RenameFileAction.java]: what about the following case:  dest prefixed with the default hdfs uri while src not, the two path are in the same cluster

Comment (qiyuangong) [inline smart-actions/src/main/java/org/smartdata/actions/hdfs/RenameFileAction.java]: That should be done after url handler is finished.

File: smart-actions/src/main/java/org/smartdata/actions/hdfs/CopyFileAction.java [hunk 3]
```diff
@@ -17,17 +17,97 @@
...
   protected void execute() throws Exception {
+    if (srcPath == null) {
+      throw new IllegalArgumentException("File parameter is missing.");
+    }
+    if (destPath == null) {
+      throw new IllegalArgumentException("Dest File parameter is missing.");
+    }
+    appendLog(
+        String.format("Action starts at %s : Read %s", Utils.getFormatedCurrentTime(), srcPath));
+    if (!dfsClient.exists(srcPath)) {
```
Comment (qiyuangong) [inline smart-actions/src/main/java/org/smartdata/actions/hdfs/CopyFileAction.java]: Need more parameters from configuration/map. Will be done later.

**Pull request merged**